test/blackbox-tests/test-cases/config/config-version.t

```diff
@@ -0,0 +1,56 @@
+Tests verifying that the config file presents accurate error or warning
+messages when a stanza is used with a version of dune which does not support
+such a stanza.
+
+Create a config file to use in the following tests. We will initialize the 
+config with a version and a stanza incompatable with that version.
+
+  $ touch dune-config
+  $ cat >dune-config <<EOF
+  > (lang dune 1.0)
+  > (cache enabled)
+  > EOF
+
+Attempt to initialize a new project while an invaild stanza due to versioning
+exists in the config.
+
+  $ dune init proj test --config-file=dune-config
+  File "$TESTCASE_ROOT/dune-config", line 2, characters 0-15:
+  2 | (cache enabled)
+      ^^^^^^^^^^^^^^^
+  Error: 'cache' is only available since version 2.0 of the dune language.
+  Please update your dune config file to have (lang dune 2.0).
+  [1]
+
+Swap out the invaild stanza for another and try again.
+
+  $ sed -i -e '$s/.*/(cache-check-probability 0.5)/' dune-config
+  $ dune init proj test --config-file=dune-config
+  File "$TESTCASE_ROOT/dune-config", line 2, characters 0-29:
+  2 | (cache-check-probability 0.5)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: 'cache-check-probability' is only available since version 2.7 of the
+  dune language. Please update your dune config file to have (lang dune 2.7).
+  [1]
+
+Swap out the invaild stanza for another and try again.
```

Comment:
Correct me if I'm misunderstanding something but past this point I don't see a purpose of the other tests. We already established that it displays the right file when it errors and it works when deeding the advice and adding a new error shows the error message again with the right error location.

I'd remove the rest of the tests as they don't exercise any more interesting edge cases (and running extra tests takes time that could be better spent).

Reply:
Your correct here, my intention was to simply cover multiple stanzas with differing supported versions of dune but 2 tests does cover this case.